pkg/controller/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,117 @@
+package bootstrap
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/openshift/machine-config-operator/pkg/controller/render"
+	"github.com/openshift/machine-config-operator/pkg/controller/template"
+	"github.com/openshift/machine-config-operator/pkg/generated/clientset/versioned/scheme"
+)
+
+// Bootstrap defines boostrap mode for Machine Config Controller
+type Bootstrap struct {
+	// dir used by template controller to render internal machineconfigs.
+	templatesDir string
+	// dir used to read pools and user defined machineconfigs.
+	manifestDir string
+}
+
+// New returns controller for bootstrap
+func New(templatesDir, manifestDir string) *Bootstrap {
+	return &Bootstrap{
+		templatesDir: templatesDir,
+		manifestDir:  manifestDir,
+	}
+}
+
+// Run runs boostrap for Machine Config Controller
+// It writes all the assets to destDir
+func (b *Bootstrap) Run(destDir string) error {
+	infos, err := ioutil.ReadDir(b.manifestDir)
```

Comment:
nit: why not just use ‘walk’?

Reply:
Didn't want to traverse subdirectories unless required.